client/src/pages/UserProfile.jsx

```diff
@@ -35,19 +38,37 @@ export default function UserProfile() {
       const { status, data } = await axios.get(`/user/${user._id}/reviews`);
       if (status >= 200 && status < 300 && mounted.current) setReviews(data.reviews);
     } catch (err) {
-      if (err?.repsonse?.status === 401) {
+      if (err?.response?.status === 401) {
         navigate('/');
       } else {
         console.error('Error fetching users reviews -->', err);
       }
     }
   };
 
+  const getUserData = async () => {
```

Comment:
The request to get user data seems unnecessary to me since it would only run when the user from context changes. If the user from context changes then we just need to update the local user state for the update form (lastName, description etc.) using the updated user object from context.